src/services/doppler-api-client.ts

```diff
@@ -0,0 +1,78 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { AppSession, DopplerAPIConnectionData } from './app-session';
+import { RefObject } from 'react';
+import { SubscriberList, SubscriberListState } from './shopify-client';
+
+export interface DopplerAPIClient {
+  getListData(idList: number, apikey: string): Promise<ResultWithoutExpectedErrors<SubscriberList>>;
+}
+
+export class HttpDopplerAPIClient implements DopplerAPIClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getDopplerAPIConnectionData(): DopplerAPIConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (
+      !connectionData ||
+      connectionData.status !== 'authenticated' ||
+      !connectionData.jwtToken ||
+      !connectionData.userData
+    ) {
+      throw new Error('Doppler API connection data is not available');
+    }
+    return { jwtToken: connectionData.jwtToken, userAccount: connectionData.userData.user.email };
+  }
+
+  private mapList(data: any): SubscriberList {
+    return {
+      name: data.name,
+      id: data.listId,
+      amountSubscribers: data.subscribersCount,
+      state:
+        data.currentStatus === SubscriberListState.ready
+          ? SubscriberListState.ready
+          : SubscriberListState.synchronizingContacts,
+    };
+  }
+
+  public async getListData(
+    listId: number,
+    apikey: string,
+  ): Promise<ResultWithoutExpectedErrors<SubscriberList>> {
+    try {
+      // eslint-disable-next-line
+      const { jwtToken, userAccount } = this.getDopplerAPIConnectionData();
+      const response = await this.axios.request({
+        method: 'GET',
+        url: `http://newapiint.fromdoppler.net/accounts/${userAccount}/lists/${listId}?api_key=${apikey}`,
+        //headers: { Authorization: `token ${jwtToken}` }, // This line will remain commented until jwtToken is enabled for dopplerAPI
+      });
+      if (response.status === 200 && response.data) {
+        return { success: true, value: this.mapList(response.data) };
+      } else {
+        return { success: false, error: response.statusText };
```

Comment:
I think that in general Doppler API stores the error message in a very specific place.

Maybe `response.title` (if it is available), see an example of Relay API Client (very similar):

https://github.com/FromDoppler/relay-webapp/blob/2b4141cb7c276d96488aa12e9d2301c7bbc9985a/src/wwwroot/interceptors/errorHandlerInterceptor.js#L29-L35

Reply:
Yes I've seen the response for error and I thought it was more descriptive statusText than title, but sure I can change it.